pennylane/devices/default_clifford.py

```diff
@@ -534,46 +752,233 @@ def _measure_state(self, tableau_simulator, circuit, global_phase):
         return state * qml.matrix(global_phase)
 
     @staticmethod
-    def _measure_density_matrix(tableau_simulator):
+    def _measure_density_matrix(tableau_simulator, wires):
         """Measure the density matrix from the state of simulator device."""
         state_vector = qml.math.array(tableau_simulator.state_vector(endian="big"))
-        return qml.math.einsum("i, j->ij", state_vector, state_vector)
+        return qml.math.reduce_dm(qml.math.einsum("i, j->ij", state_vector, state_vector), wires)
 
+    def _measure_expectation(self, tableau_simulator, meas_op, stim):
+        """Measure the expectation value with respect to the state of simulator device."""
+        # Get the observable for the expectation value measurement
+        meas_obs = meas_op.obs
+
+        coeffs, paulis = self._convert_op_to_linear_comb(meas_obs, coeffs=[], paulis=[])
+        expecs = qml.math.zeros_like(coeffs)
+        for idx, (pauli, wire) in enumerate(paulis):
+            pauli_term = ["I"] * max(np.max(list(wire)) + 1, tableau_simulator.num_qubits)
+            for op, wr in zip(pauli, wire):
+                pauli_term[wr] = op
+            stim_pauli = stim.PauliString("".join(pauli_term))
+            expecs[idx] = tableau_simulator.peek_observable_expectation(stim_pauli)
+
+        return qml.math.dot(coeffs, expecs)
+
+    def _measure_expectation_samples(self, stim_circuit, meas_op, num_shots, sample_seed):
+        """Measure the expectation value with respect to samples from simulator device."""
+        # Get the observable for the expectation value measurement
+        samples, coeffs = self._measure_observable_sample(
+            meas_op, stim_circuit, num_shots, sample_seed
+        )
+        expecs = [
+            qml.math.mean(qml.math.power([-1] * num_shots, qml.math.sum(sample, axis=1)))
+            for sample in samples
+        ]
+
+        return qml.math.dot(coeffs, expecs)
+
+    def _measure_variance(self, tableau_simulator, meas_op, stim):
+        """Measure the variance with respect to the state of simulator device."""
+        # TODO: find a better pythonic way to deal with this enable-disable
```

Comment:
😄 if it's desirable we could consider adding a contextmanager type of thing

Reply:
I added it. Works like a beauty. I can make it private if you don't want it to be public.